velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -148,16 +149,15 @@ TEST_F(DateTimeFunctionsTest, makeDate) {
                             std::optional<int32_t> day) {
     return evaluateOnce<Date>("make_date(c0, c1, c2)", year, month, day);
   };
-  Date d0;
-  parseTo("1920-01-25", d0);
-  EXPECT_EQ(makeDate(1920, 1, 25), d0);
-  EXPECT_FALSE(makeDate(std::nullopt, 7, 15).has_value());
-  EXPECT_FALSE(makeDate(2019, std::nullopt, 17).has_value());
-  EXPECT_FALSE(makeDate(2019, 8, std::nullopt).has_value());
-  EXPECT_THROW(makeDate(INT32_MAX, 12, 15), VeloxUserError);
-  EXPECT_THROW(makeDate(2021, 13, 1), VeloxUserError);
-  EXPECT_THROW(makeDate(2022, 3, 35), VeloxUserError);
-  EXPECT_THROW(makeDate(2023, 2, 29), VeloxUserError);
+  Date expectDate;
+  parseTo("1920-01-25", expectDate);
+  EXPECT_EQ(makeDate(1920, 1, 25), expectDate);
+  constexpr int32_t kMax = std::numeric_limits<int32_t>::max();
+  auto errorMessage = fmt::format("Date out of range: {}-12-15", kMax);
```

Comment:
Can you add a test with negative values?

Reply:
Notice that the code has different limits for the 'year'.

```
// Max and min year correspond to Joda datetime min and max
constexpr const int32_t kMinYear{-292275055};
constexpr const int32_t kMaxYear{292278994};

bool isValidDate(int32_t year, int32_t month, int32_t day) {
  if (month < 1 || month > 12) {
    return false;
  }
  if (year < kMinYear || year > kMaxYear) {
    return false;
  }
  if (day < 1) {
    return false;
  }
  return isLeapYear(year) ? day <= kLeapDays[month] : day <= kNormalDays[month];
}
```

Does this match Spark's semantics? Please, double check and update PR description and documentation. Please, include references to Spark's semantics.

Reply:
@pedroerp Pedro, do you have any context on this check: 

```
if (year < kMinYear || year > kMaxYear) {
```

Reply:
> @pedroerp Pedro, do you have any context on this check:

These are the limits I got from reverse engineering the Joda-java library (used in Presto Java), to strictly follow its semantic. These limits will probably be different from the Spark java implementation though.

Reply:
Hi @mbasmanova . Although in order to keep the semantics completely consistent in many details, we should set different thresholds for different engines, but in this case, the semantic inconsistency is just that the exceptions thrown may be different. `year=333,333,333` in spark will throw an `integer overflow` exception. The exception in velox is `Date out of range`. 

If Velox want to maintain completely consistent semantics, there will be a lot of redundancy in the code to implement different methods for different engines, and Velox may need to design a solution to this inconsistency at the framework level.